LandSandBoat/server
# Remove more unused BCNM code, fix rules for canLoseXp

**_I affirm:_**

## What does this pull request do?
* Removes static data from bcnm_info
* Removes bcnm_battlefield table and all references
* Fixes issue where converted battlefields did not have canLoseXp (false) applied
* Removes additional unused functions and logic from core for battlefields



## Steps to test these changes
* Enter/win/exit/lose existing BCNM
* Set BCNM allowedAreas to invalid (`set{ 4 }`) and ensure reject logic still functions
